base/util.jl

```diff
@@ -249,15 +242,13 @@ julia> @allocated rand(10^6)
 """
 macro allocated(ex)
     quote
-        let
-            local f
-            function f()
-                b0 = gc_bytes()
-                $(esc(ex))
-                gc_bytes() - b0
-            end
-            f()
-        end
+        while false; end # compiler heuristic: compile this block (alter this if the heuristic changes)
+        local b0 = Ref{Int64}(0)
+        local b1 = Ref{Int64}(0)
+        gc_bytes(b0)
```

Comment:
If this code is compiled, shouldn't it be ok for `gc_bytes()` to return a value?

Reply:
There's some code paths that'll compile but not infer this code. But anyways, it seemed nicer to be at least robust against ever including `@allocated` in the overhead.